tests/Feature/AbstractRequestTest.php

```diff
@@ -600,4 +604,54 @@ public function provideSetBodyIfNotEmpty(): array
             ['68144', true],
         ];
     }
+
+    public function testTimeout(): void
+    {
+        $request = new ConcreteRequest();
+        $defaultOptions = getProperty($request, 'guzzleOptions');
+        self::assertSame(30, $defaultOptions['timeout']);
+
+        $request->setTimeout(CarbonInterval::weeks(2));
+        // 1209600 seconds in two weeks (60 * 60 * 24 * 7 * 2)
+        self::assertSame(1209600, getProperty($request, 'guzzleOptions')['timeout']);
+    }
+
+    public function testLogsTimeout(): void
+    {
+        Storage::fake('api-logs');
+        Carbon::setTestNow('2023-02-03 00:00:00');
+        $request = new class extends ConcreteRequest {
+            protected bool $shouldLog = true;
+            public function getLogFolder(): string
+            {
+                return 'timeout';
+            }
+        };
+        $this->mockGuzzleWithTapper()->addMatch('POST', '/awesome/', function (Request $request) {
+            // Guzzle MockHandler doesn't actually observe timeouts, so we're faking behavior
+            throw new ConnectException(
+                'cURL error 28: Operation timed out after 30000 milliseconds with 0 bytes received (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://awesome-api.com/url',
+                $request,
+            );
+        });
+        try {
+            $request->sync();
+            self::fail('Should have thrown exception');
+        } catch (ConnectException $e) {
+            self::assertSame(
+                <<<LOG
+                POST https://awesome-api.com/url
+
+                Exception thrown: GuzzleHttp\Exception\ConnectException
+                cURL error 28: Operation timed out after 30000 milliseconds with 0 bytes received (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://awesome-api.com/url
+
+
+                null
```

Comment:
The way this is failing in MockHandler, AbstractRequest->requestStats doesn't get set. That's where this rogue `null` comes from.

A small part of me wants to make Tapper much fancier and give it a way to mock timeouts that sets transfer time and simplifies away the exception phrasing, but if so far we have one unit test in the library that actually needs that behavior, that's crazy overbuilding.